### DISCUSSION TITLE
[SEVI]输出CCD表格到文件的过大

### DISCUSSION THREAD

**Discussion by ruihaoliu2001** [physics-data/faq#101] [Q&A] [answered]
根据`README`描述，似乎输出的文件格式为(50000, 2048, 2048)最好，即完整描述了50000次实验中，每一次2048*2048的每一个格点的信息；
然而当将这样的表格一次输出到`hdf5`文件过程，出现了内存不够的问题(killed)；
尝试了以下办法：
1. 事先建立好该表格文件，再分50000次导入：但是发现每一步大概需要25s，50000步在10天左右，不可行；
2. 每次导入100或者500个，分500或100次：这样虽然单步速度快很多，还是会在一段时间后内存不够；

请问是否可以在表格中只记录有电子的格点的信息？
@littlewu2508 谢谢！

**Comment by littlewu2508** [comment, marked as answer]
确实这样的数据量有些大（压缩前数百 GB），但只存有电子的格点的信息不可行，否则提高部分无法完成了。

理论上分批次导入是不会出现内存不够的情况的，不清楚你的程序为什么会在后期爆内存。

我们的考察重点是对物理建模的理解、数据格式的安排。如果你觉得数据量太大电脑处理不过来（尤其是做提高部分时需要更复杂的计算），你可以适当降低实验的次数（但请不要低于1000次实验），并在实验报告中说明是由于算力不足导致的，我不会因此扣分。但如果是由于你的数据结构安排不合理，或者程序不合理导致了内存爆掉，我会扣白盒（非功能）分数。
